src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -2332,8 +2336,13 @@ export function parseExitResult(
 					commandLine += shellLaunchConfig.args.map(a => ` '${a}'`).join();
 				}
 			}
-
-			if (processState === ProcessState.KilledDuringLaunch) {
+			if (shellIntegrationAttempted) {
+				if (commandLine) {
+					message = nls.localize('launchFailed.exitCodeAndCommandLineShellIntegration', "The terminal process \"{0}\" failed to launch (exit code: {1}). Disable shell integration with `terminal.integrated.enableShellIntegration`.", commandLine, code);
+				} else {
+					message = nls.localize('launchFailed.exitCodeOnlyShellIntegration', "The terminal process failed to launch (exit code: {0}). Disable shell integration with `terminal.integrated.enableShellIntegration`.", code);
```

Comment:
These would be best as a link to open the settings UI as we could then change it to:

```
Disabling [shell integration](...) might help.
```

@rzhao271 might know how to do this?